sdks/java/core/src/main/java/org/apache/beam/sdk/io/DefaultFilenamePolicy.java

```diff
@@ -87,6 +130,17 @@ public static DefaultFilenamePolicy constructUsingStandardParameters(
   private final String shardTemplate;
   private final String suffix;
 
+  /*
+   * Checks whether given template contains enough information to form windowed file names
+   */
+  static boolean isWindowedTemplate(String template){
+    if (template != null){
+      Matcher m = WINDOWED_FORMAT_RE.matcher(template);
```

Comment:
yep.. basically detecting if user already is using windowing pattern, in which case we do not append default one... if we do not end up removing this I will clarify/rename/document further